### ISSUE TITLE
Failed to verify verification code

### ISSUE THREAD

**Issue by bliekp** [ndbroadbent/icloud_photos_downloader#56]
Output:

Two-factor authentication required. Your trusted devices are:
  0: SMS to ********29
Which device would you like to use? [0]: 
Please enter validation code: 852568
Failed to verify verification code

And yes, I'm 100% sure I have entered the correct code ;).
Any hints what is going wrong here?

**Comment by evenwebb** [comment]
Did you enter the code from the SMS text or the code from the pop up on your device?

**Comment by bliekp** [comment]
Popup. I didn't even get a SMS text...

**Comment by Armageddonas** [comment]
I am having the same problem

**Comment by DavidLoader** [comment]
It's expecting the code sent to you by SMS. You can get it to accept the code that displays natively on your iPhone by changing the code validation script in authentication.py to the following:

**Comment by ndbroadbent** [comment]
Hi @bliekp, I just ran into a similar issue. The 2FA prompt appeared on my Mac, but the only option was to send an SMS. I ignore the prompt on my Mac and entered the code from the SMS, and that worked. 

Also I just learned that "two-step authentication" and "two-factor authentication" are [different things](https://github.com/picklepete/pyicloud#two-step-and-two-factor-authentication-2sa2fa).

I think I might have changed from 2SA to 2FA, which might be causing this issue.

**Comment by ndbroadbent** [comment]
This is fixed in 90237daeafeaa81c21226d2335da9628d84eddb7. I've added a new option to enter the 2FA code that is automatically sent to your device:

```
Two-step/two-factor authentication is required.
  0: SMS to *******03
  1: Enter two-factor authentication code
Please choose an option: [0]: 1
Please enter two-factor authentication code: 123456
Great, you're all set up. The script can now be run without user interaction.
You can also set up email notifications for when the two-step authentication expires.
(Use --help to view information about SMTP options.)
Looking up all photos...
...
```

**Comment by ndbroadbent** [comment]
Thanks @DavidLoader for the tip about the empty `dict()`!
